python/cuspatial/cuspatial/core/geodataframe.py

```diff
@@ -181,31 +182,29 @@ def _slice(self: T, arg: slice) -> T:
         )
         return self.__class__(result)
 
-    def _apply_boolean_mask(self, mask) -> T:
+    def _apply_boolean_mask(self, gather_map: GatherMap) -> T:
         geo_columns, data_columns = self._split_out_geometry_columns()
-        data = data_columns._apply_boolean_mask(mask)
+        data = data_columns._apply_boolean_mask(gather_map)
```

Comment:
This one is not a `GatherMap` but a [`copy_types.BooleanMask`](https://github.com/rapidsai/cudf/blob/45763fa1f90c174ff9123b4839c729e6c0dee700/python/cudf/cudf/core/copy_types.py#L110).

Reply:
Note that cudf's [`_apply_boolean_mask`](https://github.com/rapidsai/cudf/blob/45763fa1f90c174ff9123b4839c729e6c0dee700/python/cudf/cudf/core/indexed_frame.py#L3017) now also accepts a `keep_index` keyword argument

Reply:
cuspatial should really just try to avoid depending on cudf internals...

Reply:
I guess that is difficult if you're subclassing though :(